test/gen_rpc/remote_functional_SUITE.erl

```diff
@@ -0,0 +1,269 @@
+        safe_cast_inexistent_node/1,
+        client_inactivity_timeout/1,
+        server_inactivity_timeout/1]).
+
+%%% Auxiliary functions for test cases
+-export([interleaved_call_proc/3, interleaved_call_executor/1]).
+
+%%% ===================================================
+%%% CT callback functions
+%%% ===================================================
+all() ->
+    {exports, Functions} = lists:keyfind(exports, 1, ?MODULE:module_info()),
+    [FName || {FName, _} <- lists:filter(
+                               fun ({module_info,_}) -> false;
+                                   ({all,_}) -> false;
+                                   ({init_per_suite,1}) -> false;
+                                   ({end_per_suite,1}) -> false;
+                                   ({interleaved_call_proc,3}) -> false;
+                                   ({interleaved_call_executor,1}) -> false;
+                                   ({_,1}) -> true;
+                                   ({_,_}) -> false
+                               end, Functions)].
+
+init_per_suite(Config) ->
+    %% Starting Distributed Erlang on local node
+    {ok, _Pid} = gen_rpc_test_helper:start_target(?NODE),
+    %% Setup application logging
+    ?set_application_environment(),
+    %% Starting the application locally
+    {ok, _MasterApps} = application:ensure_all_started(?APP),
+    ok = ct:pal("Started [functional] suite with master node [~s]", [node()]),
+    Config.
+
+end_per_suite(_Config) ->
+    ok.
+
+init_per_testcase(client_inactivity_timeout, Config) ->
+    ok = start_slave(),
+    ok = ?restart_application(),
+    ok = application:set_env(?APP, client_inactivity_timeout, infinity),
+    Config;
+init_per_testcase(server_inactivity_timeout, Config) ->
+    ok = start_slave(),
+    ok = ?restart_application(),
+    ok = application:set_env(?APP, server_inactivity_timeout, infinity),
+    Config;
+init_per_testcase(_OtherTest, Config) ->
+    ok = start_slave(),
+    Config.
+
+end_per_testcase(client_inactivity_timeout, Config) ->
+    ok = slave:stop(?SLAVE),
+    ok = ?restart_application(),
+    Config;
+end_per_testcase(server_inactivity_timeout, Config) ->
+    ok = slave:stop(?SLAVE),
+    ok = ?restart_application(),
+    Config;
+
+end_per_testcase(_OtherTest, Config) ->
+    ok = slave:stop(?SLAVE),
+    Config.
+
+
+%%% ===================================================
+%%% Test cases
+%%% ===================================================
+%% Test supervisor's status
+supervisor_black_box(_Config) ->
+    ok = ct:pal("Testing [supervisor_black_box]"),
+    true = erlang:is_process_alive(whereis(gen_rpc_server_sup)),
+    true = erlang:is_process_alive(whereis(gen_rpc_acceptor_sup)),
+    true = erlang:is_process_alive(whereis(gen_rpc_client_sup)),
+    ok.
+
+%% Test main functions
+call(_Config) ->
+    ok = ct:pal("Testing [call]"),
+    {_Mega, _Sec, _Micro} = gen_rpc:call(?SLAVE, os, timestamp).
+
+%call_anonymous_function(_Config) ->
```

Comment:
I see your point and I also experienced it when writing exrpc! I think since RPC also fails, I should document it. But in the meantime, I think you should remove the commented tests.